lib/js/jsfetch.nim

```diff
@@ -0,0 +1,120 @@
+## - Fetch for the JavaScript target: https://developer.mozilla.org/docs/Web/API/Fetch_API
+from httpcore import HttpMethod
+export HttpMethod
+
```

Comment:
can you add a top-level `runnableExamples:` illustrating a few key API's ?
if you want to disable it from CI you can write:
```nim
runnableExamples:
  if defined(nimJsfetchEnableTest):
    body...
```

if instead of when ensures body compiles; and this allows uses to test code easily

Reply:
Should `fetchMethodToCstring` be public?, maybe is useful for third party library developers. :thinking:

Reply:
right now it's deadcode because private + unused in your module, which is odd; making it public would be fine but needs more thought on that API (eg: why not raise in `else:` branch instead silently accepting invalid input?)